pkg/apis/core/types.go

```diff
@@ -2887,6 +2887,29 @@ type PodSpec struct {
 	// All topologySpreadConstraints are ANDed.
 	// +optional
 	TopologySpreadConstraints []TopologySpreadConstraint
+	// Specifies the OS on which pod intends to run. This field is needed during the API server admission time to
+	// distinguish among various node OSes in a heterogeneous Kubernetes cluster.
```

Comment:
"on which pod intends to run" is unclear whether it refers to the OS of the containers, or the OS of the node. This specified the OS of the containers in the pod, *not* necessarily the OS of the node, right?

Reply:
Drop the "This field is needed during the API server admission time ... " sentence. This field informs API validation, admission, and kubelet behavior. We don't need to narrowly scope the use in the field doc.

Reply:
This seems like a good place to enumerate the additional validation restrictions on the pod spec for each OS

Reply:
(also, copy these doc changes to the v1/types.go file)